server-spi/src/main/java/org/keycloak/models/GroupProvider.java

```diff
@@ -205,6 +204,8 @@ default GroupModel createGroup(RealmModel realm, String name, GroupModel toParen
      */
     GroupModel createGroup(RealmModel realm, String id, String name, GroupModel toParent);
 
+    Set<String> getTopLevelGroupsIdsStream(RealmModel realm, String search, Boolean exact, Integer firstResult, Integer maxResults);
```

Comment:
Please keep it consistent with other methods and return Stream<String>. Also please add javadoc.

Reply:
I've removed this method in favor of not breaking backward compatibility. We can look at it as a follow-up and see how to improve all this back-and-forth when dealign with streams later.